### PR TITLE
feat: add support for bsc and polygon mpcv2

### PR DESCRIPTION
**Pull request by dpkjnr**
Bsc recovery(staging):

> https://testnet.bscscan.com/tx/0xcd4d31ff94379a9d3e39f32a8f16fb1300b1ae959d36e34993908c930dced471
> wallet-id: 66569c49faf995c6fb11b361407532a6

Polygon recovery(test):
> https://amoy.polygonscan.com/tx/0x6a0e01bf0d7198f63a1ae5029116ae4cce8ca6756263443f2595cc0e5b820aeb
> wallet-id: 665698c441d778aa653ebbf0bef8c955

### CODE REVIEW

File: scripts/build-icons.js [hunk 1]
```diff
@@ -4,39 +4,7 @@ const cheerio = require('cheerio');
...
-  'bld',
-  'hash',
-  'sei',
-  'zeta',
-  'coreum',
-  'arbeth',
-  'opeth',
-  'hbar',
-  'algo',
+  'bsc',
```
Comment (sachushaji) [inline scripts/build-icons.js:7]: Is this intentional to remove all the coins except bsc?

File: scripts/build-icons.js [hunk 2]
```diff
@@ -163,10 +131,12 @@ function template(paths) {
...
 }
 
-fs.writeFileSync(
-  path.resolve(
-    __dirname,
-    '../src/components/CryptocurrencyIcon/CryptocurrencyIcon.tsx'
-  ),
-  template(paths)
-);
+console.log(template(paths));
```
Comment (sachushaji) [inline scripts/build-icons.js:134]: May need to uncomment this?

File: src/helpers/config.ts [hunk 3]
```diff
@@ -439,13 +447,23 @@ export const allCoinMetas: Record<string, CoinMetadata> = {
...
+    isTssSupported: true,
+    ApiKeyProvider: 'api-amoy.polygonscan.com',
+  },
+  tbsc: {
+    Title: 'TBSC',
+    Description: 'BNB Smart Chain Testnet',
+    Icon: 'bsc',
+    value: 'tbsc',
+    isTssSupported: true,
+    ApiKeyProvider: 'api-amoy.polygonscan.com',
```
Comment (sachushaji) [inline src/helpers/config.ts:459]: Not sure if this is the right provider for bsc

File: src/helpers/index.ts [hunk 4]
```diff
@@ -41,7 +41,19 @@ export function getEthLikeRecoveryChainId(
   coinName: string,
   bitGoEnvironment: string
 ) {
-  return coinName === 'ethw' ? 10001 : bitGoEnvironment === 'prod' ? 1 : 17000;
+  if(coinName === 'ethw') {
+    return 10001;
```
Comment (sachushaji) [inline src/helpers/index.ts:45]: We could get this info from statics

File: src/containers/NonBitGoRecoveryCoin/NonBitGoRecoveryCoin.tsx [hunk 5]
```diff
@@ -337,6 +336,10 @@ function Form() {
     case 'tarbeth':
     case 'opeth':
     case 'topeth':
+    case 'polygon':
+    case 'tpolygon':
+    case 'bsc':
+    case 'tbsc':
```
Comment (sachushaji) [inline src/containers/NonBitGoRecoveryCoin/NonBitGoRecoveryCoin.tsx:342]: For bsc and tbsc, don't support eip1559 on the chain right?, Have you made changes to accomodate that?

Comment (dpkjnr) [reply src/containers/NonBitGoRecoveryCoin/NonBitGoRecoveryCoin.tsx:342]: bsc now supports eip1559 so I used the common recover function. We can add eip1559 support for bsc in wallet platform too as a separate task

File: src/helpers/config.ts (hunk 3 above)
Comment (dpkjnr) [reply src/helpers/config.ts:459]: I think it is not being used, removing it

**Review by sachushaji — APPROVED** [review]

**Pull request merged**